src/test/java/duck/storage/StorageTest.java

```diff
@@ -0,0 +1,164 @@
+package duck.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Scanner;
+
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.RepetitionInfo;
+
+import duck.data.TaskList;
+import duck.data.exception.DuckException;
+import duck.data.task.Task;
+import duck.data.task.ToDo;
+
+/**
+ * Tests for the {@link Storage} class to verify file operations like writing and appending tasks.
+ */
+public class StorageTest {
+
+    /**
+     * Tests if the {@link Storage#writeTasks(TaskList)} method overwrites the file content.
+     * The test is repeated 3 times to ensure consistency.
+     *
+     * @throws DuckException If there is an issue with the Duck-specific exception handling.
+     * @throws IOException   If there is an issue with file operations.
+     */
+    @RepeatedTest(3)
+    public void writeTasks_repeatedly_shouldOverWrite() throws DuckException, IOException {
```

Comment:
Your test cases are quite extensive, and I like that you've chosen to repeat the tests multiple times to ensure consistent results. Great job!